session.go

```diff
@@ -570,11 +608,32 @@ func (s *Session) initialized() bool {
 	return initialized
 }
 
+func (s *Session) Ready() error {
+	s.sessionStateMu.RLock()
+	err := ErrSessionNotReady
+	if s.isInitialized || s.initErr != nil {
+		err = s.initErr
+	}
+	s.sessionStateMu.RUnlock()
+	return err
+}
+
+func (s *Session) WaitUntilReady() error {
+	err, ok := <-s.readyCh
+	if !ok {
+		return nil
+	}
+	return err
```

Comment:
When you close it it returns nil.
It is better to make `s.readyCh` be `chan struct{}` and return s.initErr